lib/queue.js

```diff
@@ -279,7 +279,51 @@ export default class Queue {
     });
   }
 
+  // Incoming messages get sent either now or later based on send_after
   sendMessage(data, delay, metadata = {}) {
+    if (data.send_after) {
+      const now = new Date();
+
+      // Parse send_after as a date
+      data.send_after = new Date(data.send_after)
+
+      if (data.send_after.getTime() < now.getTime()) {
+        return this.sendMessageNow(data, delay, metadata = {})
+      } else {
+        return this.sendMessageLater(data, metadata = {})
+      }
+    }
+    else {
+      return this.sendMessageNow(data, delay, metadata = {})
+    }
+  }
+
+  sendMessageLater(data, metadata = {}) {
```

Comment:
Similar to the `WorkerPool` concept in here (although we stopped using it), it would be nice to expose some method like `checkForDelayedMessages(interval)` that just uses some cron library internally to check on the provided interval.